### ISSUE TITLE
Improve "Link existing booking" feature

### ISSUE THREAD

**Issue by tomasr8** [indico/indico#6568]
When you click on "Link existing booking", you are redirected to the bookings page:
![image](https://github.com/user-attachments/assets/86e072ac-ac22-4d91-8cca-ffd40481bda2)

There are two things we could improve here:
- Currently, the bookings page uses the current date. It'd be better if it used the date of the event (or the first day in case of multi-day events)
- The bookings page currently shows all bookings which makes it hard to find a booking which can be linked. We should filter the bookings using the linked event to only show relevant bookings.

cc @Moliholy @OmeGak

**Comment by alittle60** [comment]
The above enhancement would be very helpful to users at Perimeter Institute.

Thanks very much,

cc @Moliholy @OmeGak

**Comment by tomako** [comment]
Hey @tomasr8 , could you elaborate a bit your 2nd suggestion? I would use event's venue room (if that's set) for filtering and show only relevant bookings, but maybe your idea was something else.

**Comment by ThiefMaster** [comment]
I think it's about hiding (or clearly graying out) any booking which you cannot link...

**Comment by alittle60** [comment]
Exactly, show only bookings that you can actually link.

Thanks!

**Comment by tomako** [comment]
My question was rather related to "what relevant booking is" but I think I got the answer :)
So only the relevant bookings should be displayed, which are bookings
* in the event's date interval
* in the same room as the event's venue room (if it's set)

(Sidenote: actually the calendar view respects the event start date, but this is missing from the list view)

**Comment by tomasr8** [comment]
> in the event's date interval

FWIW we might want to relax the overlap rules for when a booking can be linked to an event, but that can be done separately.

**Comment by ThiefMaster** [comment]
Yes please, if the booking is longer than he event it still makes sense to allow it for linking, because often you keep some buffer around the actual event times when reserving a room so you can take overruns or setting up the room into account

**Comment by tomako** [comment]
Well, my initial idea was tweaking the current list a bit by scrolling automatically to the first booking in the event's date interval. It wouldn't hide anything (you can move up and down) only respects the event's date. Similarly to the calendar view. Maybe this way?

**Comment by ThiefMaster** [comment]
I'd rather pass the linking data to the API endpoint and then only load relevant stuff... otherwise, especially for bookings far in the future, it's a mess because of lazy-loading.

**Comment by tomako** [comment]
Yeah, that's also true. 
Then the goal is to display existing bookings from the beginning of the event and don't bother with the end date. Got it.

**Comment by OmeGak** [comment]
> Yes please, if the booking is longer than he event it still makes sense to allow it for linking, because often you keep some buffer around the actual event times when reserving a room so you can take overruns or setting up the room into account

I would keep this as a separate development, honestly. I can imagine things becoming quite more complex. This is just a quick list of questions that came to mind right away. 

How long should the buffer be? One day? Two days? What if an someone needs three days instead? Could there be a global setting that configures this buffer? Could it be configured per room?Should it perhaps be possible to book and link any booking regardless of how far the event is from it with a warning? If so, how would this be reflected in the UI? And so on and so on.

I will, in fact, create a new issue to track and discuss this feature, as it is part of what needs to be done for Perimeter Institute.

**Comment by OmeGak** [comment]
> I'd rather pass the linking data to the API endpoint and then only load relevant stuff... otherwise, especially for bookings far in the future, it's a mess because of lazy-loading.

Taking into account that there will be a "Show all bookings" toggle as per https://github.com/indico/indico/issues/6811#issuecomment-2751644471, the API endpoint should always return all bookings. We could still limit the results based on room filtering if there is one configured in the event settings. What do you think?

---

As a summary, this is what I think it could still be addressed in this issue:
- Currently, the bookings page uses the current date. It'd be better if it used the date of the event (or the first day in case of multi-day events)
- [MAYBE] Filter API results by room?
